analyzers/src/SonarAnalyzer.Common/Rules/ArrayPassedAsParamsBase.cs

```diff
@@ -39,14 +39,19 @@ public abstract class ArrayPassedAsParamsBase<TSyntaxKind, TArgumentNode> : Sona
         context.RegisterNodeAction(Language.GeneratedCodeRecognizer, c =>
         {
             if (LastArgumentIfArrayCreation(c.Node) is { } lastArgument
-                && IsParamParameter(c.SemanticModel, c.Node, lastArgument))
+                && ParameterSymbol(c.SemanticModel, c.Node, lastArgument) is { IsParams: true } param
+                && !IsJaggedArrayParam(param))
             {
-                c.ReportIssue(Diagnostic.Create(rule, lastArgument.GetLocation()));
+                c.ReportIssue(rule, lastArgument.GetLocation());
             }
         }, ExpressionKinds);
 
-    private bool IsParamParameter(SemanticModel model, SyntaxNode invocation, TArgumentNode argument) =>
+    private IParameterSymbol ParameterSymbol(SemanticModel model, SyntaxNode invocation, TArgumentNode argument) =>
         model.GetSymbolInfo(invocation).Symbol is IMethodSymbol methodSymbol
         && Language.MethodParameterLookup(invocation, methodSymbol).TryGetSymbol(argument, out var param)
-        && param.IsParams;
+            ? param
+            : null;
+
+    private static bool IsJaggedArrayParam(IParameterSymbol param) =>
+        param.Type is IArrayTypeSymbol { ElementType: IArrayTypeSymbol };
```

Comment:
`param.Type` is always `IArrayTypeSymbol`, correct? I wonder if a direct cast would be more explicit here. Then again... I like the pattern-matching syntax :D

Reply:
I swear I replied but I don't see the message now for some reason. Anyway not always, but in this particular implementation, we are checking previously that the parameter is an array. So yes, that part will always be true. But I would leave the pattern-matching because it's quite neat IMO

Reply:
It seems like it affects coverage:
![image](https://github.com/SonarSource/sonar-dotnet/assets/20153123/51a7a038-bcc3-4b8a-9751-419c50d5b33e)
I'm ok with accepting it as defensive programming.